### DISCUSSION TITLE
Candle pattern recognition retuning values

### DISCUSSION THREAD

**Discussion by sribna** [LupeCode/phpTraderNative#10] [Help] [answered]
Hello
Thanks for your effort writing this package

When I call candlestick pattern recognition functions, they return either false or an array of calculated data. How to interpret the arrays?

For example this

```
array(1) {
  [11]=>
  float(0)
}
```

Key is candle index, right? What is 0?

According to your code https://github.com/LupeCode/phpTraderNative/blob/e77d8e8fec778023e995d10db9b52f4f44bd1075/source/TALib/Core/PatternRecognition.php#L682

I assume 0 - not found, 100 - bullish candle, -100 - bearish candle. I'm I right?

**Comment by jb-lopez** [comment, marked as answer]
Hi @sribna ,
Thank you for your question. You are correct.

I understand the confusion. Even though this library provides the same results as the Trader extension, the PHP docs do not say what the results are either.

0 is not found, 100 is bullish, and -100 is bearish. Some candles are only bearish or bullish, and some can be both.

**Comment by jabarrioss** [comment]
I have no idea of how to use this package, wish me luck 🤞

**Reply by jb-lopez** [reply]
Hi @jabarrioss. Let me know if you have any questions.
